app/views/jobs/index.html.erb

```diff
@@ -5,19 +5,25 @@
 <table>
   <thead>
   <tr>
+    <th> Status</th>
     <th> Job Name</th>
     <th> Counts </th>
     <th colspan="3"></th>
   </tr>
   </thead>
-
   <tbody>
     <% @groups.count.each do |k,v|%>
+      <% joblist = Job.where("name= ?", k).order("created_at DESC") %>
       <tr>
+        <% if joblist.first["created_at"] + joblist.first["frequency"] >= Date.today  && joblist.collect {|x| x.status == "success"}.all? %>
```

Comment:
I couldn't trigger this failure (even if I entered -1 as the frequency) so I did some digging.  In the rails console if you evaluate the parts you're comparing (joblist.first["created_at"] + joblist.first["frequency"]) you see that it's a different type than Date.today.  It might work if you use Time.now instead of Date.today.